Static/DataSource.swift

```diff
@@ -151,8 +151,7 @@ public class DataSource: NSObject {
                 tableView.insertSections(IndexSet(integersIn: range), with: animation)
             } else {
                 // Remove sections
-                let start = oldCount - 1
-                let range: Range<IndexSet.Element> = start..<(start - delta)
+                let range: Range<IndexSet.Element> = newCount..<(newCount - delta)
```

Comment:
Reading through this, why the move from start to newCount? Given the difference from index vs count, is this `-1` no longer necessary?

Reply:
With the additional tests added, if the sections are being cleared, the range passed into `deleteSections` was incorrect and causing a crash when going from 2 sections to 0 sections.

The range created with `start` is `1..<3`, which would crash with `attempt to delete section 2, but there are only 2 sections before the update`.

With `newCount`, the correct range is generated, `0..<2`, and both sections are deleted with passing tests.